app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -188,15 +189,29 @@ def setup_agent(instance)
 
       docker_username = docker_auth.userid
       docker_password = docker_auth.password
-      command_line = "docker login"
-      command_line << " #{docker_image}"
-      command_line << " -u #{docker_username} -p #{docker_password}"
-      ssh.perform_commands([command_line])
+      login_params = {
+        nil  => docker_image,
+        '-u' => docker_username,
+        '-p' => docker_password
+      }
+      login_cmd = AwesomeSpawn.build_command_line("sudo docker login", login_params)
+
+      ssh.perform_commands([login_cmd])
     end
 
     # run docker image
-    command_line = "sudo docker run -d --restart=always -v /dev:/host_dev -v #{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml --privileged #{docker_image}"
-    ssh.perform_commands([command_line])
+    run_params = {
+      '-d'           => nil,
+      '--restart='   => 'always',
+      # different format of -v to avoid duplicate keys
+      :v             => '/dev:/host_dev',
+      '-v'           => "#{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml",
```

Comment:
If you are trying to pass keys twice, then just use an associative array instead of a hash.  Also prefer long-form option names if possible for better readability.

```ruby
run_params = [
  '-d',
  ['--restart=', 'always'],
  ['-v', '/dev:/host_dev'],
  ['-v', "#{WORK_DIR}/config.yml:#{WORK_DIR}/config.yml"],
  '--privileged',
  docker_image
}
```

Reply:
Thank, I'll change it.